src/VisualStudio/Core/Def/Implementation/ProjectSystem/AbstractProject.cs

```diff
@@ -134,7 +134,7 @@ internal abstract partial class AbstractProject : ForegroundThreadAffinitizedObj
 
             this.RunningDocumentTable = (IVsRunningDocumentTable4)serviceProvider.GetService(typeof(SVsRunningDocumentTable));
 
-            var displayName = hierarchy != null && hierarchy.TryGetName(out var name) ? name : projectSystemName;
+            var displayName = projectSystemName;
```

Comment:
I'm sure that what I did here isn't right....But, basically I want to reuse the logic in `FindMatchingProject` that gets a project name from `IVsContainedLanguageProjectNameProvider` and then filters the projects on the project name and hierarchy. 

The issue with this is that the `IWorkspaceProjectContext` APIs don't really allow me to set the display name, as it's ignored here when I provide a hierarchy.